templates/nextroute/main.go

```diff
@@ -47,8 +47,8 @@ func solver(
 	}
 	last := solutions.Last()
 
-	output := nextroute.Format(ctx, options, solver, last)
-	output.Statistics.Result.Custom = nextroute.DefaultCustomResultStatistics(last)
+	output := factory.Format(ctx, options, solver, last)
```

Comment:
This is a breaking change and we should not introduce it. I don't see the benefit of shielding the format function, it is just statistics calculation and it actually makes sense to keep it public to reference when someone is implementing a custom formatter.